app/src/main/java/com/example/mvicompose/presentation/CharactersViewModel.kt

```diff
@@ -31,16 +33,15 @@ class CharactersViewModel @ViewModelInject constructor (
     val state: StateFlow<CharactersViewState>
         get() = _state
 
-
     init {
         intentsChannel
             .asFlow()
             .map { intent -> intent.toAction() }
             .process()
-            .scan(DefaultState as CharactersViewState) { state, result ->
-                with (stateMachine) { state reduce result }
+            .scan(defaultAsOption) { state, result ->
```

Comment:
Can we change this name for defaultStateAsOption? It would be more descriptive. 😉